### DISCUSSION TITLE
Framework X CLI 👀

### DISCUSSION THREAD

**Discussion by JustSteveKing** [clue/framework-x#223] [Ideas]
I had an idea when talking to a few other developers and thought I would post it here.

What do people think about creating a Framework X CLI? Something that could be used for starting the PHP server or getting information about the ongoing server process etc.

```bash
# This would start the PHP Server.
php x server
```

```bash
# This would display information about the running PHP Server
php x info
```

The info command could list how many processes are running, the PHP version, the Framework X version, the ReactPHP components being used, and perhaps the amount of async processes that are currently running.

**Comment by SimonFrings** [comment]
Hey @JustSteveKing, thanks for bringing this up :+1:

I can see this being a useful feature, but I'm not sure if this fits so well with Framework X. In order to properly introduce this, we'd also have to answer a lot of questions, like:

- If executing `php x server`, which running instance should this talk to and what if there are multiple instances running?
- Which protocol would we use for communication and who is allowed to communicate with this protocol?
- What happens when running this in production?
- How can this be configured?
- What if an attacker sends a "shutdown" command?
- ...

With all this in mind I'm not sure ho much of a benefit this currently brings compared to other desired features we have in our pipeline. We're currently working towards a `v0.17.0` which restructures our existing handlers and lays the foundation to add even more handlers in an upcoming release (maybe `v0.18.0`).

 I think this is a valid suggestion and this might be a useful addition, but I think this is something we'll look into in the future. If you (or anyone else) needs a feature like this (more) urgently, you can also reach out to us or help us prioritizing by [becoming as sponsor](https://github.com/sponsors/reactphp) ❤️
